parser/files/reading.go

```diff
@@ -0,0 +1,463 @@
+	ctx, cancel := context.WithCancel(context.Background())
+	gzipCommand := exec.CommandContext(ctx, gzipPath, "-d", "-c")
+
+	// tell the subprocess to read from the given stream
+	gzipCommand.Stdin = fileHandle
+
+	// return/ pipe the output back out to the caller
+	pipeR, err := gzipCommand.StdoutPipe()
+	if err != nil {
+		cancel() // essentially a no-op.  makes the linter happy tho.
+		return reader, fileHandle.Close, err
+	}
+
+	var cmdStdErr bytes.Buffer
+	gzipCommand.Stderr = &cmdStdErr
+
+	if err := gzipCommand.Start(); err != nil {
+		cancel() // essentially a no-op.  makes the linter happy tho.
+		return reader, fileHandle.Close, err
+	}
+
+	// update the closer to kill the subprocess in addition to closing the file descriptor
+	closer = func() error {
+		// kill the subprocess, any errors will come out on the read side or during Wait
+		cancel()
+		// close the file that was passed in
+		errFile := fileHandle.Close()
+		// wait for the subprocess to finish out
+		errProc := gzipCommand.Wait()
+
+		// add StdErr to the process error if the command returned a nonzero code
+		if errProc != nil && cmdStdErr.Len() > 0 {
+			errProc = fmt.Errorf("%s: %s", errProc.Error(), cmdStdErr.String())
+		}
+
+		// handle return errors up
+		if errProc != nil && errFile != nil {
+			return fmt.Errorf("%s; %s", errProc.Error(), errFile.Error())
+		}
+		if errProc != nil {
+			return errProc
+		}
+		if errFile != nil {
+			return errFile
+		}
+		return nil
+	}
+
+	return pipeR, closer, nil
+}
+
+// scanHeader scans the comment lines out of a bro file and returns a
+// BroHeader object containing the information. NOTE: This has the side
+// effect of advancing the fileScanner so that fileScanner.Text() will
+// return the first log entry in the file.
+func scanTSVHeader(fileScanner *bufio.Scanner) (*BroHeader, error) {
+	toReturn := new(BroHeader)
+	for fileScanner.Scan() {
+		if fileScanner.Err() != nil {
+			break
+		}
+		if len(fileScanner.Bytes()) < 1 {
+			continue
+		}
+		//On the comment lines
+		if fileScanner.Bytes()[0] == '#' {
+			line := strings.Fields(fileScanner.Text())
+			switch line[0][1:] {
+			case "separator":
+				var err error
+				toReturn.Separator, err = strconv.Unquote("\"" + line[1] + "\"")
+				if err != nil {
+					return toReturn, err
+				}
+			case "set_separator":
+				toReturn.SetSep = line[1]
+			case "empty_field":
+				toReturn.Empty = line[1]
+			case "unset_field":
+				toReturn.Unset = line[1]
+			case "fields":
+				toReturn.Names = line[1:]
+			case "types":
+				toReturn.Types = line[1:]
+			case "path":
+				toReturn.ObjType = line[1]
+			}
+		} else {
+			//We are done parsing the comments
+			break
+		}
+	}
+
+	if len(toReturn.Names) != len(toReturn.Types) {
+		return toReturn, errors.New("name / type mismatch")
+	}
+	return toReturn, nil
+}
+
+func mapZeekHeaderToParseType(header *BroHeader, broDataFactory func() pt.BroData, logger *log.Logger) (ZeekHeaderIndexMap, error) {
+	broData := broDataFactory()
```

Comment:
I kept this one as `broData` since the type was kept as `pt.BroData` from previous versions. I've added an issue to the tracker to generally rename the references from Bro to Zeek (#693).